### PR TITLE
Fallback to auth-provider header case

### PR DESCRIPTION
**Pull request by gilliardmacedo**
I'm proposing a solution for https://github.com/redwoodjs/redwood/issues/8367

### CODE REVIEW

File: packages/api/src/auth/index.ts [hunk 1]
```diff
@@ -6,10 +6,10 @@ import type { Decoded } from './parseJWT'
 export type { Decoded }
 
 // This is shared by `@redwoodjs/web`
-const AUTH_PROVIDER_HEADER = 'auth-provider'
+const AUTH_PROVIDER_HEADER = 'Auth-Provider'
 
 export const getAuthProviderHeader = (event: APIGatewayProxyEvent) => {
-  return event?.headers[AUTH_PROVIDER_HEADER]
+  return event?.headers[AUTH_PROVIDER_HEADER] || event?.headers[AUTH_PROVIDER_HEADER.toLowercase()]
```
Comment (jtoar) [inline packages/api/src/auth/index.ts:12]:
```suggestion
  return event?.headers[AUTH_PROVIDER_HEADER] || event?.headers[AUTH_PROVIDER_HEADER.toLowerCase()]
```

File: packages/api/src/auth/index.ts [hunk 2]
```diff
@@ -6,10 +6,10 @@ import type { Decoded } from './parseJWT'
 export type { Decoded }
 
 // This is shared by `@redwoodjs/web`
-const AUTH_PROVIDER_HEADER = 'auth-provider'
+const AUTH_PROVIDER_HEADER = 'Auth-Provider'
 
 export const getAuthProviderHeader = (event: APIGatewayProxyEvent) => {
-  return event?.headers[AUTH_PROVIDER_HEADER]
+  return event?.headers[AUTH_PROVIDER_HEADER] || event?.headers[AUTH_PROVIDER_HEADER.toLowerCase()]
```
Comment (jtoar) [inline packages/api/src/auth/index.ts:12]:
```suggestion
  return event?.headers[AUTH_PROVIDER_HEADER] ?? event?.headers[AUTH_PROVIDER_HEADER.toLowerCase()]
```

**Review by jtoar — APPROVED** [review]
I'd rather handle this in a holistic way (i.e. make sure all header casing is normalized, etc.), but since this doesn't do any harm (it's just an additional check) and @gilliardmacedo needs this and contributed a fix, let's get this out in a RC and then a patch

**Comment by standup75** [comment]
I am probably just confused by the process here, I see this in 5.2.2-rc.0 and the main branch but not in the last release (5.2.3), can someone tell me why?

**Comment by jtoar** [comment]
This PR is a special case where the team didn't feel comfortable releasing it in a stable version yet. I got the RC out for @gilliardmacedo in the meantime. Are you affected too @standup75?

**Comment by standup75** [comment]
Sorry for the late reply. It does affect me too, but Gilliard and I are working on the same project :)
Not a problem though we upgraded to 5.2.3 and added the behavior in our graphql function handler for now. Not too ugly.

**Comment by jtoar** [comment]
@standup75 don't worry, and I just talked to @thedavidprice—we'll release this in the next minor, sound good?

**Comment by standup75** [comment]
> @standup75 don't worry, and I just talked to @thedavidprice—we'll release this in the next minor, sound good?

Awesome thank you

**Comment by jtoar** [comment]
@standup75, @gilliardmacedo we just released a fix for this in [v5.2.4](https://github.com/redwoodjs/redwood/releases/tag/v5.2.4). We went with a slightly different implementation but it should cover your use case still: https://github.com/redwoodjs/redwood/pull/8474.

**Comment by standup75** [comment]
Awesome thank you

**Pull request merged**